mypy/types.py

```diff
@@ -254,13 +254,16 @@ class AnyType(Type):
     def __init__(self,
                  implicit: bool = False,
                  from_unimported_type: bool = False,
+                 explicit: bool = False,
                  line: int = -1,
                  column: int = -1) -> None:
         super().__init__(line, column)
         # Was this Any type was inferred without a type annotation?
         self.implicit = implicit
         # Does this come from an unfollowed import? See --disallow-any=unimported option
         self.from_unimported_type = from_unimported_type
+        # Does this Any come from an explicit type annotation?
+        self.explicit = explicit
```

Comment:
Yes, `explicit` is not always the opposite of `implicit`, which is confusing.

`implicit` refers to situations when `Any` was inferred because a variable or parameter was not annotated.
`explicit` refers to situations when there was an explicit `Any` annotation. 

However, if this `Any` comes, for instance, from an unimported type, both `explicit` and `implicit` will be `False`.

It seems to me that `implicit` has a wrong name. It is currently used for `--disallow-untyped-defs` to see if this `Any` comes from a parameter or variable with no explicit type annotation. 

I think instead of tweaking comments we should rename `implicit`. How do you feel about calling it `from_unannotated`?